src/binaries/nrmc.c

```diff
@@ -732,83 +690,56 @@ static struct client_cmd commands[] = {
         {0, 0},
 };
 
+void nrmc_list_commands()
+{
+	fprintf(stdout, "\nAvailable Commands:\n");
+	for (size_t i = 0; commands[i].name != NULL; i++)
+		fprintf(stdout, "%s\n", commands[i].name);
+}
+
 int main(int argc, char *argv[])
 {
-	int c;
-	int option_index = 0;
+	int err;
+	nrm_tools_args_t args;
+	nrm_init(NULL, NULL);
+	nrm_log_init(stderr, "nrmc");
 
-	while (1) {
-		c = getopt_long(argc, argv, short_options, long_options,
-		                &option_index);
-		if (c == -1)
-			break;
-		switch (c) {
-		case 0:
-			break;
-		case 'h':
-			ask_help = 1;
-			break;
-		case 'p':
-			errno = 0;
-			pub_port = strtol(optarg, NULL, 0);
-			assert(errno == 0);
-			break;
-		case 'q':
-			log_level = NRM_LOG_QUIET;
-			break;
-		case 'r':
-			errno = 0;
-			rpc_port = strtol(optarg, NULL, 0);
-			assert(errno == 0);
-			break;
-		case 'u':
-			upstream_uri = optarg;
-			break;
-		case 'V':
-			ask_version = 1;
-			break;
-		case ':':
-			fprintf(stderr, "nrmc: missing argument\n");
-			exit(EXIT_FAILURE);
-		case '?':
-			fprintf(stderr, "nrmc: invalid option: %s\n",
-			        argv[optind - 1]);
-			exit(EXIT_FAILURE);
-		default:
-			fprintf(stderr, "nrmc: this should not happen\n");
-			exit(EXIT_FAILURE);
-		}
+	args.progname = "nrmc";
+	args.flags = 0;
+
+	err = nrm_tools_parse_args(argc, argv, &args);
+	if (err < 0) {
+		nrm_log_error("errors during argument parsing\n");
+		exit(EXIT_FAILURE);
 	}
 
 	/* remove the parsed part */
-	argc -= optind;
-	argv = &(argv[optind]);
+	argc -= err;
+	argv = &(argv[err]);
 
-	if (ask_help) {
-		print_help();
+	if (args.ask_help) {
+		nrm_tools_print_help(&args);
 		exit(EXIT_SUCCESS);
 	}
```

Comment:
Come to think of it, shouldn't printed help always include the list of commands?